tools/compare_i18n.js

```diff
@@ -0,0 +1,113 @@
+// Script to compare keys between translation JSON files across all projects
+const fs = require('fs');
+const path = require('path');
+
+// Function to flatten a JSON object
+function flattenObject(obj, prefix = '') {
+  return Object.keys(obj).reduce((acc, k) => {
+    const pre = prefix.length ? prefix + '.' : '';
+    if (typeof obj[k] === 'object' && obj[k] !== null && !Array.isArray(obj[k])) {
+      Object.assign(acc, flattenObject(obj[k], pre + k));
+    } else {
+      acc[pre + k] = obj[k];
+    }
+    return acc;
+  }, {});
+}
+
+// Track if any errors are found
+let errorsFound = false;
+
+// Function to compare two JSON files
+function compareJsonFiles(frPath, enPath, projectName, dirType) {
+  try {
+    const frJson = require(frPath);
+    const enJson = require(enPath);
+
+    const flatFr = flattenObject(frJson);
+    const flatEn = flattenObject(enJson);
+
+    const missingInEn = Object.keys(flatFr).filter(key => !flatEn.hasOwnProperty(key));
+    const missingInFr = Object.keys(flatEn).filter(key => !flatFr.hasOwnProperty(key));
+
+    console.log(`\n=== Project: ${projectName} (${dirType}) ===`);
+
+    if (missingInEn.length > 0) {
+      console.log(`Keys missing in English file (${enPath}):`);
+      missingInEn.forEach(key => console.log(`  - ${key}`));
+      errorsFound = true;
+    } else {
+      console.log('No keys missing in English file.');
+    }
+
+    if (missingInFr.length > 0) {
+      console.log(`Keys missing in French file (${frPath}):`);
+      missingInFr.forEach(key => console.log(`  - ${key}`));
+      errorsFound = true;
+    } else {
+      console.log('No keys missing in French file.');
+    }
+  } catch (error) {
+    console.error(`Error comparing files for ${projectName} (${dirType}):`, error.message);
+    errorsFound = true;
+  }
+}
+
+// Function to check a specific i18n directory
+function checkI18nDirectory(i18nDir, projectName, dirType) {
+  if (fs.existsSync(i18nDir)) {
+    const frPath = path.join(i18nDir, 'fr.json');
+    const enPath = path.join(i18nDir, 'en.json');
+
+    // Check if both files exist
+    if (fs.existsSync(frPath) && fs.existsSync(enPath)) {
+      compareJsonFiles(frPath, enPath, projectName, dirType);
+      return true;
+    } else {
+      console.log(`\n=== Project: ${projectName} (${dirType}) ===`);
+      console.log('Incomplete translation files:');
+      if (!fs.existsSync(frPath)) console.log('  - fr.json missing');
+      if (!fs.existsSync(enPath)) console.log('  - en.json missing');
+      errorsFound = true;
+      return false;
+    }
+  }
+  return false;
+}
+
+// Main function to scan directories
+function scanProjects() {
+  const projectsDir = path.resolve('../ui/ui-frontend/projects'); // projects/ directory
```

Comment:
Pour pouvoir être exécuté depuis n'importe quel dossier :
```suggestion
  const projectsDir = path.resolve(__dirname, '../ui/ui-frontend/projects'); // projects/ directory
```